eng/pipelines/templates/jobs/vmr-build.yml

```diff
@@ -309,25 +330,56 @@ jobs:
         docker run --rm $dockerVolumeArgs -w /vmr $dockerEnvArgs ${{ parameters.container }} ./build.sh --source-only $poisonArg --test $(additionalBuildArgs) /p:SmokeTestConsoleVerbosity=detailed
       displayName: Run Tests
 
-  # Don't use CopyFiles@2 as it encounters permissions issues because it indexes all files in the source directory graph.
-  - script: |
-      set -x
-
-      targetFolder=$(Build.StagingDirectory)/BuildLogs/
-      mkdir -p ${targetFolder}
-
-      cd "$(sourcesPath)"
-      find artifacts/ -type f -name "*.binlog" -exec cp {} --parents -t ${targetFolder} \;
-      find artifacts/ -type f -name "*.log" -exec cp {} --parents -t ${targetFolder} \;
-      find artifacts/prebuilt-report/ -exec cp {} --parents -t ${targetFolder} \;
-      find src/ -type f -name "*.binlog" -exec cp {} --parents -t ${targetFolder} \;
-      find src/ -type f -name "*.log" -exec cp {} --parents -t ${targetFolder} \;
-      find test/ -type f -name "*.binlog" -exec cp {} --parents -t ${targetFolder} \;
-      find test/ -type f -name "Updated*.diff" -exec cp {} --parents -t ${targetFolder} \;
-      find test/ -type f -name "Updated*.txt" -exec cp {} --parents -t ${targetFolder} \;
-    displayName: Prepare BuildLogs staging directory
-    continueOnError: true
-    condition: succeededOrFailed()
+  - ${{ if eq(parameters.targetOS, 'windows') }}:
+
+    # Don't use CopyFiles@2 as it encounters permissions issues because it indexes all files in the source directory graph.
+    - powershell: |
+        function CopyWithRelativeFolders($sourcePath, $targetFolder, $filter) {
+          Get-ChildItem -Path $sourcePath -Filter $filter -Recurse | ForEach-Object { 
+            $targetPath = Join-Path $targetFolder (Resolve-Path -Relative $_.FullName)
+            New-Item -ItemType Directory -Path (Split-Path -Parent $targetPath) -Force | Out-Null
+            Copy-Item $_.FullName -Destination $targetPath -Force 
+          }
+        }
+
+        $targetFolder = "$(Build.StagingDirectory)/BuildLogs/"
+        New-Item -ItemType Directory -Path $targetFolder -Force | Out-Null
+
+        cd "$(sourcesPath)"
+
+        CopyWithRelativeFolders "artifacts/"                 $targetFolder "*.binlog"
+        CopyWithRelativeFolders "artifacts/"                 $targetFolder "*.log"
+        CopyWithRelativeFolders "artifacts/prebuilt-report/" $targetFolder
+        CopyWithRelativeFolders "src/"                       $targetFolder "*.binlog"
+        CopyWithRelativeFolders "src/"                       $targetFolder "*.log"
```

Comment:
Unrelated but is it possible to look under `src/*/artifacts/logs` only? The current glob brings a test artifact in runtime illink in. Same q for the Unix counterpart.

Reply:
hm not easily I think, please file an issue so we can follow up